packages/fast-react-utilities/README.md

```diff
@@ -0,0 +1,66 @@
+# FAST React Utilities
+
+A set of general purpose React utilities.
+
+## Installation
+
+`npm i --save @microsoft/fast-react-utilities`
+
+## Exports
+### Hooks
+#### `useTimeout(callback: () => void, delay: number | null, memoKeys?: any[]): void`
+A React hook to declaritivly invoke a timeout function. The callback be invoked once after `delay` - measured in miliseconds. Once the timeout is invoked, no other timeout will be registerd unless duration changes.
+
+To force a new timeout to be registed with a previous duration, supply *new* values to the `memoKeys`. This is similar to how React's `useEffect` works.
+
+##### Single execution of callback
+```js
+function FancyButton() {
+    // Execute a callback invoked after 200ms. Callback will only be called once (unless duration changes)
+    useTimeout(() => {
+        alert("I'm a button")
+    }, 200);
+
+    return <button>hello world</button>
+}
+```
+##### Execute callback whenever prop is changed
+```js
+function FancyButton() {
+    // Execute a callback 200ms after render. A new callback will be registered
+    // when props.value changes
+    useTimeout((props) => {
+        alert("I'm a button")
+    }, 200, [props.value]);
+
+    return <button>hello world</button>
+}
+```
+
+##### Execute callback every render
+```js
+function FancyButton() {
+    // Execute a callback 200ms after every render. If render happens before the delay,
+    // the previous render's timeout will be canceled.
+    useTimeout(() => {
+        alert("I'm a button")
+    }, 200, [Symbol()]);
+
+    return <button>hello world</button>
+}
+```
+
+#### `useTransitionState(active: boolean, duration: number | [number, number]): TransitionStates`
+A React hook used to track the state of a transition based on initial and incoming `active` values. `useTransitionState` will return a value from the `TransitionStates` object, representing "inactive", "active", "activating", and "deactivating". When the incoming `active` argument changes from one render to the next, the return value will change to one of the *middle* states, "activating" or "deactivating" depending on if the value is going from `false`->`true` or `true`->`false`. After the supplied duration, the hook will invoke a `setState` and return the state represented by the incoming `active` value.
+
+When two duration values are provided, the first will be used when `active` goes from `false`->`true` and the second will be used when `active` goes from `true`->`false`.
+
+```js
+function FancyDialog(props) {
+    const state = useTransitionState(props.visible); 
+
+    // eg, apply classes dependent on if the dialog is
+    // resting in an active / inactive state or if it is in the
+    // middle of trasitioning to active / inactive

```

Comment:
Could we provide one example of this either in place of or in addition to the comment here?